tools/restatectl/src/commands/cluster/provision.rs

```diff
@@ -56,13 +51,14 @@ pub struct ProvisionOpts {
 }
 
 async fn cluster_provision(
-    connection_info: &ConnectionInfo,
+    connection: &ConnectionInfo,
     provision_opts: &ProvisionOpts,
 ) -> anyhow::Result<()> {
-    let node_address = provision_opts
-        .address
-        .clone()
-        .unwrap_or_else(|| connection_info.cluster_controller.clone());
+    if connection.addresses.len() != 1 {
+        anyhow::bail!("Only one address must be provided for provision");
+    }
```

Comment:
If we are using the configured `addresses`, then it is probably fine to also continue if there are multiple `addresses` configured. We would just take the first one. Maybe we log that we are using this address when multiple ones are present.